pkg/storaged/filesystem/utils.jsx

```diff
@@ -35,21 +35,65 @@ import { mounting_dialog } from "./mounting-dialog.jsx";
 
 const _ = cockpit.gettext;
 
-export function is_mounted(client, block) {
+/* is mounted with an entry in fstab */
+export function is_mounted(client, block, subvol) {
     const block_fsys = client.blocks_fsys[block.path];
     const mounted_at = block_fsys ? block_fsys.MountPoints : [];
-    const config = block.Configuration.find(c => c[0] == "fstab");
-    if (config && config[1].dir.v) {
-        let dir = decode_filename(config[1].dir.v);
-        if (dir[0] != "/")
-            dir = "/" + dir;
+    const [, dir] = get_fstab_config(block, false, subvol);
+    if (dir) {
         return mounted_at.map(decode_filename).indexOf(dir) >= 0;
     } else
         return null;
 }
 
-export function get_fstab_config(block, also_child_config) {
-    return get_fstab_config_with_client(client, block, also_child_config);
+export function get_fstab_config(block, also_child_config, subvol) {
+    return get_fstab_config_with_client(client, block, also_child_config, subvol);
+}
+
+/**
+ * Find all subvolumes with the given mountpoint in fstab without using
+ * get_fstab_config_with_client as it doesn't return multiple entries given the
+ * same block device.
+ */
+function find_subvols_for_mount_point(client, mount_point, subvol) {
+    const subvols = [];
+    const match = c => {
+        let self = false;
+
+        if (c[0] != "fstab")
+            return false;
+
+        const dir = decode_filename(c[1].dir.v);
+        const opts = decode_filename(c[1].opts.v);
+
+        const subvol_match = opts.match(/subvol=(?<subvol>[\w\\/]+)/);
+        if (subvol_match && subvol_match.groups.subvol == subvol.pathname)
+            self = true;
```

Comment:
This must also take into account the volume, not just the subvolume.

If I make two btrfs, one on /dev/sda and one on /dev/sdb, then both have a "/" subvolume, but they are not the same, of course.

Reply:
So thinking of a test for this it would be:

```
mkfs.btrfs /dev/sda
mkfs.btrfs /dev/sdb
```

fstab:
```
/dev/sda /mnt/sda btrfs subvol=/
/dev/sdb /mnt/sdb btrfs subvol=/
```

 However, this only matches when a subvol and a mount point reaches the condition if `dir == mount_point && !self`. The `!self` confuses me a bit, shouldn't that be `self`.

Reply:
Maybe the function name is a bit deceiving and needs to be fixed. Looking back at this code it needs to detect if anything else already has a mount point for this subvolume. Anyway I think the point still stands, let's make a test for this and watch it fail :+1:

Reply:
This function finds all fstab entries of btrfs volumes for `mount_point`, but ignores those for the given subvolume.

`find_blocks_for_mount_point` does that job as well, and could be improved to correctly deal with subvols, by adding a subvol check to its `is_self` function.

Reply:
You also need to test btrfs volumes with more than one device. I still get bogus "mount point is already in use", with this:

```
mkfs.btrfs /dev/sda /dev/sdb
```

fstab:
```
UUID=blahbla /mnt/butter auto subvol=/
```

Cockpit finds /mnt/butter also on /dev/sdb.